Neos.EventSourcedContentRepository/Classes/Domain/ValueObject/CommandResult.php

```diff
@@ -109,20 +103,27 @@ protected static function blockProjector(DomainEvents $events, ProcessedEventsAw
                     $ids .= '   ' . $p->getIdentifier();
                 }
 
-                throw new \RuntimeException('TIMEOUT while waiting for events: ' . $ids, 1550232279);
+                throw new \RuntimeException(sprintf('TIMEOUT while waiting for event(s) %s for projector "%s"', $ids, get_class($projector)), 1550232279);
             }
         }
     }
 
     private function filterPublishedEventsByListener(string $listenerClassName): DomainEvents
     {
-        $eventClassNames = $this->eventListenerLocator->getEventClassNamesByListenerClassName($listenerClassName);
+        $eventStoreId = $this->eventToListenerMappingProvider->getEventStoreIdentifierForListenerClassName($listenerClassName);
+        $listenerMappings = $this->eventToListenerMappingProvider->getMappingsForEventStore($eventStoreId)->filter(static function (EventToListenerMapping $mapping) use ($listenerClassName) {
+            return $mapping->getListenerClassName() === $listenerClassName;
+        });
+        $eventClassNames = [];
+        foreach ($listenerMappings as $mapping) {
+            $eventClassNames[$mapping->getEventClassName()] = true;
+        }
```

Comment:
^ This whole part is more cumbersome than it should be.. Maybe we can extend the mapping classes a bit to make things like this easer?

Reply:
fine for now to merge it IMHO. But... can't we move all this code to a Flow\CompileStatic part? as it does not depend on any runtime state AFAICS.

Reply:
The result of this method depends on `$this->publishedEvents`.. But I could imagine some convenience methods on `EventToListenerMappings` – since those are compiled.
Maybe even `getEventClassNamesByListenerClassName()` that we had before via the `EventListenerLocator`..

Reply:
yeah the result of the full method depends on $this->publishedEvents; but the first part (building the lookup) seems to be static. right?